### ISSUE TITLE
twisted.cred documentation is unclear about CredentialChecker implementations

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3954]
|<img alt="rlotun's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| rlotun reported|
|-|-|
|Trac ID|trac#3954|
|Type|defect|
|Created|2009-08-06 15:18:25Z|

I read this paragraph from the twisted cred documentation tutorial:

```
This is an object implementing ICredentialsChecker which resolves some Credentials to an avatar ID. Some examples of CredentialChecker implementations would be: InMemoryUsernamePassword, ApacheStyleHTAccessFile, UNIXPasswordDatabase, SSHPublicKeyDatabase. A credential checker stipulates some requirements of the credentials it can check by specifying a credentialInterfaces attribute, which is a list of interfaces. Credentials passed to its requestAvatarId method must implement one of those interfaces.
```

And it led me to believe that `ApacheStyleHTAccessFile` actually existed. Of course, I think the intention was that it _could_ be a possible implementation. Anyway, this needs to be reworded a bit more clearly.

Attachments:

* [cred-doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5fa/5faa7e4e1f7c253df37d3e999be83b8d3e4e9609/86ab9e9e30bf3a542e46076956cca576638b4eaf.patch) (1572 bytes) - added by rlotun on 2009-08-06 15:50:00Z - Patch to alleviate some confusion in the twisted.cred doc
* [plain-english-checkers.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5fa/5faa7e4e1f7c253df37d3e999be83b8d3e4e9609/01eaa38a4fef60b0313f450a4d59a8ced3ad3116.patch) (1121 bytes) - added by exarkun on 2009-08-19 22:13:52Z - A different wording change proposal

<details><summary>Searchable metadata</summary>

```
trac-id__3954 3954
type__defect defect
reporter__rlotun rlotun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1249571905000000 1249571905000000
changetime__1251732579000000 1251732579000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="rlotun's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|rlotun commented|
|-|-|

Urg, I formatting a little messed up:

```
This is an object implementing ICredentialsChecker which resolves some Credentials to an 
avatar ID. Some examples of CredentialChecker implementations would be: 
InMemoryUsernamePassword, ApacheStyleHTAccessFile, UNIXPasswordDatabase, 
SSHPublicKeyDatabase. A credential checker stipulates some requirements of the credentials it can check by specifying a credentialInterfaces attribute, which is a list 
of interfaces. Credentials passed to its requestAvatarId method must implement one of 
those interfaces.
}```{
This is an object implementing ICredentialsChecker which resolves some Credentials to an 
avatar ID. Some examples of CredentialChecker implementations would be: 
InMemoryUsernamePassword, ApacheStyleHTAccessFile, UNIXPasswordDatabase, 
SSHPublicKeyDatabase. A credential checker stipulates some requirements of the credentials it can check by specifying a credentialInterfaces attribute, which is a list 
of interfaces. Credentials passed to its requestAvatarId method must implement one of 
those interfaces.
}}

**Comment by twisted-trac** [comment]
|<img alt="rlotun's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|rlotun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|ivank commented|
|-|-|

I agree that it was confusing. The patch looks good to me. No words were lost, and it links to the right place.

Someone with SVN access could merge; I'm leaving the review keyword so that this doesn't get lost.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @exarkun|
|-|-|

I find the use of theoretical class names here to not contribute much.  I'm going to attach another proposal.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Looks good to me, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [27263]) Apply plain-english-checkers.patch

Author: exarkun
Reviewer: thijs
Fixes: [#3954](https://github.com/twisted/twisted/issues/3954)

Adjust the wording in the text which gives examples of credentials checkers so
that it does not seem to imply the existence of a number of non-existent classes.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
